### ISSUE TITLE
ModuleNotFoundError: No module named 'triqs.utility'; 'triqs' is not a package

### ISSUE THREAD

**Issue by Vahid999** [TRIQS/triqs#971]
I load the triqs-3.1.1 module on our ComputeCanada cluster. This module has the triqs_dft_tools-3.1.0 extension. I have prepared the Wannier90 seedname_hr.dat and the seedname.inp files. When I run "python triqs.py" where triqs.py is

`from triqs_dft_tools.converters import Wannier90Converter

Converter = Wannier90Converter(seedname='Ca2N.sa', bloch_basis=False, rot_mat_type='hloc_diag', add_lambda=None)`

I get the following error:

`Traceback (most recent call last):
  File "triqs.py", line 1, in <module>
    from triqs_dft_tools.converters import Wannier90Converter
  File "/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/MPI/gcc10/openmpi4/triqs/3.1.1/lib/python3.8/site-packages/triqs_dft_tools/__init__.py", line 23, in <module>
    from .sumk_dft import SumkDFT
  File "/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/MPI/gcc10/openmpi4/triqs/3.1.1/lib/python3.8/site-packages/triqs_dft_tools/sumk_dft.py", line 29, in <module>
    import triqs.utility.dichotomy as dichotomy
  File "/lustre07/scratch/vaskarpo/DMFT/wannier/triqs.py", line 1, in <module>
    from triqs_dft_tools.converters import Wannier90Converter
  File "/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/MPI/gcc10/openmpi4/triqs/3.1.1/lib/python3.8/site-packages/triqs_dft_tools/converters/__init__.py", line 26, in <module>
    from .wien2k import Wien2kConverter
  File "/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/MPI/gcc10/openmpi4/triqs/3.1.1/lib/python3.8/site-packages/triqs_dft_tools/converters/wien2k.py", line 29, in <module>
    from .converter_tools import *
  File "/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/MPI/gcc10/openmpi4/triqs/3.1.1/lib/python3.8/site-packages/triqs_dft_tools/converters/converter_tools.py", line 26, in <module>
    import triqs.utility.mpi as mpi
ModuleNotFoundError: No module named 'triqs.utility'; 'triqs' is not a package`

I would appreciate any suggestions as to how to avoid this error.

Thanks,
Vahid

**Comment by the-hampel** [comment]
Hi @Vahid999 ,

this is very strange. I actually also have an ComputeCanada cluster access (might be just some small testing machine, but it has all the official modules). So I went ahead and I could not reproduce your problem. Here is what I did: 
```
[xyz@login1 ~]$ module load StdEnv/2020 gcc/10.3.0
[xyz@login1 ~]$ module load triqs/3.1.1
[xyz@login1 ~]$ module list

Currently Loaded Modules:
  1) CCconfig          3) imkl/2020.1.217 (math)   5) gcccore/.10.3.0 (H)   7) ucx/1.9.0          9) pmix/3.2.3         11) flexiblas/3.0.4      13) clang/13.0.1   (t)     15) hdf5-mpi/1.12.1 (io)  17) python/3.10.2 (t)  19) ipykernel/2022a           21) triqs/3.1.1
  2) gentoo/2020 (S)   4) StdEnv/2020     (S)      6) gcc/10.3.0      (t)   8) libfabric/1.12.1  10) openmpi/4.1.1 (m)  12) boost/1.76.0    (t)  14) fftw-mpi/3.3.9 (math)  16) libffi/3.3            18) mpi4py/3.1.3  (t)  20) scipy-stack/2022a (math)

...

[lect15@login1 ~]$ python
Python 3.10.2 (main, Feb  4 2022, 19:11:32) [GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import triqs.utility
>>> from triqs_dft_tools.converters import Wannier90Converter
Warning: could not identify MPI environment!
Starting serial run at: 2024-11-18 08:05:58.950498
```

Right at the bottom I both tried to load directly `triqs.utility` and also load the `Wannier90Converter`. This was successful. Can you show me your loaded modules on the machines and the run: 
```
python -c "from triqs_dft_tools.converters import Wannier90Converter"
```

Best,
Alex

**Comment by Vahid999** [comment]
Hi Alex,

Thank you for your suggestions. I loaded the modules as you said on Narval. The modules are:

***@***.***:~/scratch/DMFT$ module list
Currently Loaded Modules:
  1) CCconfig                 6) gcccore/.10.3.0 (H)  11) ucx/1.9.0             16) clang/13.0.1    (t)     21) mpi4py/3.1.3      (t)
  2) gentoo/2020     (S)      7) gcc/10.3.0      (t)  12) libfabric/1.12.1      17) fftw-mpi/3.3.9  (math)  22) ipykernel/2022a
  3) imkl/2020.1.217 (math)   8) flexiblas/3.0.4      13) pmix/3.2.3            18) hdf5-mpi/1.12.1 (io)    23) scipy-stack/2022a (math)
  4) StdEnv/2020     (S)      9) blis/0.8.1           14) openmpi/4.1.1    (m)  19) libffi/3.3              24) triqs/3.1.1
  5) mii/1.1.2               10) hwloc/2.4.0          15) boost/1.76.0     (t)  20) python/3.10.2   (t)

And the python commands:

***@***.***:~/scratch/DMFT$ python
Python 3.10.2 (main, Feb  4 2022, 19:10:35) [GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import triqs.utility
>>> from triqs_dft_tools.converters import Wannier90Converter

Warning: could not identify MPI environment!
Starting serial run at: 2024-11-18 06:43:46.519327
>>> python -c "from triqs_dft_tools.converters import Wannier90Converter"
  File "<stdin>", line 1
    python -c "from triqs_dft_tools.converters import Wannier90Converter"
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
SyntaxError: invalid syntax


Regards,
Vahid

On Nov 18, 2024, at 4:12 AM, Alexander Hampel ***@***.***> wrote:

CAUTION: The Sender of this email is not from within Dalhousie.


Hi @Vahid999<https://github.com/Vahid999> ,

this is very strange. I actually also have an ComputeCanada cluster access (might be just some small testing machine, but it has all the official modules). So I went ahead and I could not reproduce your problem. Here is what I did:

***@***.*** ~]$ module load StdEnv/2020 gcc/10.3.0
***@***.*** ~]$ module load triqs/3.1.1
***@***.*** ~]$ module list

Currently Loaded Modules:
  1) CCconfig          3) imkl/2020.1.217 (math)   5) gcccore/.10.3.0 (H)   7) ucx/1.9.0          9) pmix/3.2.3         11) flexiblas/3.0.4      13) clang/13.0.1   (t)     15) hdf5-mpi/1.12.1 (io)  17) python/3.10.2 (t)  19) ipykernel/2022a           21) triqs/3.1.1
  2) gentoo/2020 (S)   4) StdEnv/2020     (S)      6) gcc/10.3.0      (t)   8) libfabric/1.12.1  10) openmpi/4.1.1 (m)  12) boost/1.76.0    (t)  14) fftw-mpi/3.3.9 (math)  16) libffi/3.3            18) mpi4py/3.1.3  (t)  20) scipy-stack/2022a (math)

...

***@***.*** ~]$ python
Python 3.10.2 (main, Feb  4 2022, 19:11:32) [GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import triqs.utility
>>> from triqs_dft_tools.converters import Wannier90Converter
Warning: could not identify MPI environment!
Starting serial run at: 2024-11-18 08:05:58.950498


Right at the bottom I both tried to load directly triqs.utility and also load the Wannier90Converter. This was successful. Can you show me your loaded modules on the machines and the run:

python -c "from triqs_dft_tools.converters import Wannier90Converter"


Best,
Alex

**Comment by the-hampel** [comment]
Okay this part looks absolutely correct:
> Python 3.10.2 (main, Feb  4 2022, 19:10:35) [GCC 9.3.0] on linux
> Type "help", "copyright", "credits" or "license" for more information.
> import triqs.utility
> from triqs_dft_tools.converters import Wannier90Converter
> Warning: could not identify MPI environment!
> Starting serial run at: 2024-11-18 06:43:46.519327

So how did you run the script? Any different than this? Can you try this again just on the login node. Just to check if it starts. You can terminate it immediately if there are no errors. There needs to be some difference to your actual script execution, because otherwise everything seems to be fine.

**Comment by Vahid999** [comment]
Hi Alex,

Something seems to be wrong with ComputeCanada because the steps I took last time produced a different outcome. I load all the modules, run python and then try the following which throws an error (it did not show any error last time):

>>> import triqs.utility
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/lustre07/scratch/vaskarpo/DMFT/wannier/triqs.py", line 1, in <module>
    from triqs_dft_tools.converters import Wannier90Converter
  File "/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/MPI/gcc10/openmpi4/triqs/3.1.1/lib/python3.10/site-packages/triqs_dft_tools/__init__.py", line 23, in <module>
    from .sumk_dft import SumkDFT
  File "/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/MPI/gcc10/openmpi4/triqs/3.1.1/lib/python3.10/site-packages/triqs_dft_tools/sumk_dft.py", line 29, in <module>
    import triqs.utility.dichotomy as dichotomy
ModuleNotFoundError: No module named 'triqs.utility'; 'triqs' is not a package

The list of the loaded modules is

Currently Loaded Modules:
  1) CCconfig                 6) gcccore/.10.3.0 (H)  11) ucx/1.9.0             16) clang/13.0.1    (t)     21) mpi4py/3.1.3      (t)
  2) gentoo/2020     (S)      7) gcc/10.3.0      (t)  12) libfabric/1.12.1      17) fftw-mpi/3.3.9  (math)  22) ipykernel/2022a
  3) imkl/2020.1.217 (math)   8) flexiblas/3.0.4      13) pmix/3.2.3            18) hdf5-mpi/1.12.1 (io)    23) scipy-stack/2022a (math)
  4) StdEnv/2020     (S)      9) blis/0.8.1           14) openmpi/4.1.1    (m)  19) libffi/3.3              24) triqs/3.1.1
  5) mii/1.1.2               10) hwloc/2.4.0          15) boost/1.76.0     (t)  20) python/3.10.2   (t)

I will contact the system administrator to see if there are issues with some of these modules.

Thanks,
Vahid



On Nov 18, 2024, at 4:12 AM, Alexander Hampel ***@***.***> wrote:

CAUTION: The Sender of this email is not from within Dalhousie.


Hi @Vahid999<https://github.com/Vahid999> ,

this is very strange. I actually also have an ComputeCanada cluster access (might be just some small testing machine, but it has all the official modules). So I went ahead and I could not reproduce your problem. Here is what I did:

***@***.*** ~]$ module load StdEnv/2020 gcc/10.3.0
***@***.*** ~]$ module load triqs/3.1.1
***@***.*** ~]$ module list

Currently Loaded Modules:
  1) CCconfig          3) imkl/2020.1.217 (math)   5) gcccore/.10.3.0 (H)   7) ucx/1.9.0          9) pmix/3.2.3         11) flexiblas/3.0.4      13) clang/13.0.1   (t)     15) hdf5-mpi/1.12.1 (io)  17) python/3.10.2 (t)  19) ipykernel/2022a           21) triqs/3.1.1
  2) gentoo/2020 (S)   4) StdEnv/2020     (S)      6) gcc/10.3.0      (t)   8) libfabric/1.12.1  10) openmpi/4.1.1 (m)  12) boost/1.76.0    (t)  14) fftw-mpi/3.3.9 (math)  16) libffi/3.3            18) mpi4py/3.1.3  (t)  20) scipy-stack/2022a (math)

...

***@***.*** ~]$ python
Python 3.10.2 (main, Feb  4 2022, 19:11:32) [GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import triqs.utility
>>> from triqs_dft_tools.converters import Wannier90Converter
Warning: could not identify MPI environment!
Starting serial run at: 2024-11-18 08:05:58.950498


Right at the bottom I both tried to load directly triqs.utility and also load the Wannier90Converter. This was successful. Can you show me your loaded modules on the machines and the run:

python -c "from triqs_dft_tools.converters import Wannier90Converter"


Best,
Alex

**Comment by Vahid999** [comment]
Hi Alex,

The issue was caused by the file triqs.py in the working directory. I renamed it to triqs.in and ran “python triqs.in” and now it runs fine. Maybe triqs.py is also a file internal to the source code of triqs.

Best,
Vahid

On Nov 18, 2024, at 3:50 PM, Vahid Askarpour ***@***.***> wrote:

Hi Alex,

Something seems to be wrong with ComputeCanada because the steps I took last time produced a different outcome. I load all the modules, run python and then try the following which throws an error (it did not show any error last time):

>>> import triqs.utility
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/lustre07/scratch/vaskarpo/DMFT/wannier/triqs.py", line 1, in <module>
    from triqs_dft_tools.converters import Wannier90Converter
  File "/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/MPI/gcc10/openmpi4/triqs/3.1.1/lib/python3.10/site-packages/triqs_dft_tools/__init__.py", line 23, in <module>
    from .sumk_dft import SumkDFT
  File "/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/MPI/gcc10/openmpi4/triqs/3.1.1/lib/python3.10/site-packages/triqs_dft_tools/sumk_dft.py", line 29, in <module>
    import triqs.utility.dichotomy as dichotomy
ModuleNotFoundError: No module named 'triqs.utility'; 'triqs' is not a package

The list of the loaded modules is

Currently Loaded Modules:
  1) CCconfig                 6) gcccore/.10.3.0 (H)  11) ucx/1.9.0             16) clang/13.0.1    (t)     21) mpi4py/3.1.3      (t)
  2) gentoo/2020     (S)      7) gcc/10.3.0      (t)  12) libfabric/1.12.1      17) fftw-mpi/3.3.9  (math)  22) ipykernel/2022a
  3) imkl/2020.1.217 (math)   8) flexiblas/3.0.4      13) pmix/3.2.3            18) hdf5-mpi/1.12.1 (io)    23) scipy-stack/2022a (math)
  4) StdEnv/2020     (S)      9) blis/0.8.1           14) openmpi/4.1.1    (m)  19) libffi/3.3              24) triqs/3.1.1
  5) mii/1.1.2               10) hwloc/2.4.0          15) boost/1.76.0     (t)  20) python/3.10.2   (t)

I will contact the system administrator to see if there are issues with some of these modules.

Thanks,
Vahid



On Nov 18, 2024, at 4:12 AM, Alexander Hampel ***@***.***> wrote:

CAUTION: The Sender of this email is not from within Dalhousie.


Hi @Vahid999<https://github.com/Vahid999> ,

this is very strange. I actually also have an ComputeCanada cluster access (might be just some small testing machine, but it has all the official modules). So I went ahead and I could not reproduce your problem. Here is what I did:

***@***.*** ~]$ module load StdEnv/2020 gcc/10.3.0
***@***.*** ~]$ module load triqs/3.1.1
***@***.*** ~]$ module list

Currently Loaded Modules:
  1) CCconfig          3) imkl/2020.1.217 (math)   5) gcccore/.10.3.0 (H)   7) ucx/1.9.0          9) pmix/3.2.3         11) flexiblas/3.0.4      13) clang/13.0.1   (t)     15) hdf5-mpi/1.12.1 (io)  17) python/3.10.2 (t)  19) ipykernel/2022a           21) triqs/3.1.1
  2) gentoo/2020 (S)   4) StdEnv/2020     (S)      6) gcc/10.3.0      (t)   8) libfabric/1.12.1  10) openmpi/4.1.1 (m)  12) boost/1.76.0    (t)  14) fftw-mpi/3.3.9 (math)  16) libffi/3.3            18) mpi4py/3.1.3  (t)  20) scipy-stack/2022a (math)

...

***@***.*** ~]$ python
Python 3.10.2 (main, Feb  4 2022, 19:11:32) [GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import triqs.utility
>>> from triqs_dft_tools.converters import Wannier90Converter
Warning: could not identify MPI environment!
Starting serial run at: 2024-11-18 08:05:58.950498


Right at the bottom I both tried to load directly triqs.utility and also load the Wannier90Converter. This was successful. Can you show me your loaded modules on the machines and the run:

python -c "from triqs_dft_tools.converters import Wannier90Converter"


Best,
Alex

**Comment by the-hampel** [comment]
Ah! I did not read this carefully enough in your first post. If you name your script triqs.py it will give you a lot of problems. If you name the script triqs.py and do inside `from triqs.utility import *` it will try to search inside of triqs.py for a module called utility, i.e. a namespace triqs is reserved by the name of the script itself. If that collides with some site package it is of course a problem. That is actually not triqs specific. I did not pay attention to that detail in your first post and only read the error message... Happy that this is resolved now and someone spotted the mistake. 
Best,
Alex
